include/real/real.hpp

```diff
@@ -92,25 +92,35 @@ namespace boost {
              *
              * @throws boost::real::invalid_string_number exception if string doesn't represent a valid number
              */
-            real(const std::string& number) {
-                auto [integer_part, decimal_part, exponent, positive] = exact_number<>::number_from_string(number);
-
-                if ((int)(decimal_part.length() + integer_part.length()) <= exponent) {
-                    this->_real_p = std::make_shared<real_data<T>>(real_explicit<T>(integer_part, decimal_part, exponent, positive));
-                } else {
-                    int zeroes = decimal_part.length() + integer_part.length() - exponent;
-                    std::string denominator = "1";
-                    for (int i = 0; i<zeroes; ++i)
-                        denominator = denominator + "0";
-
-                    // source of inefficiency. copying, casting.
-                    std::string numerator = (std::string) std::string(integer_part).c_str() + (std::string) std::string(decimal_part);
-                    if (!positive)
-                        numerator = "-" + numerator;
-                    std::shared_ptr<real_data<T>> lhs = std::make_shared<real_data<T>>(real_explicit<T>(numerator));
-                    std::shared_ptr<real_data<T>> rhs = std::make_shared<real_data<T>>(real_explicit<T>(denominator));
-    
-                    this->_real_p  = std::make_shared<real_data<T>>(real_operation(lhs, rhs, OPERATION::DIVISION));
+            real(const std::string& number, std::string type = "explicit") {
+                if(type=="explicit"){
+                    auto [integer_part, decimal_part, exponent, positive] = exact_number<>::number_from_string(number);
+
+                    if ((int)(decimal_part.length() + integer_part.length()) <= exponent) {
+                        this->_real_p = std::make_shared<real_data<T>>(real_explicit<T>(integer_part, decimal_part, exponent, positive));
+                    } else {
+                        int zeroes = decimal_part.length() + integer_part.length() - exponent;
+                        std::string denominator = "1";
+                        for (int i = 0; i<zeroes; ++i)
+                            denominator = denominator + "0";
+
+                        // source of inefficiency. copying, casting.
+                        std::string numerator = (std::string) std::string(integer_part).c_str() + (std::string) std::string(decimal_part);
+                        if (!positive)
+                            numerator = "-" + numerator;
+                        std::shared_ptr<real_data<T>> lhs = std::make_shared<real_data<T>>(real_explicit<T>(numerator));
```

Comment:
Having it as a shared pointer doesnt open the door to someone changing the value of one of the parameters and breaking what we tought was a number? 
We should check that we have all the const in place for preventing that from happening.

Reply:
Actually, I just copy pasted this part from previous code. This constructor of explicit number was not made by me, I just put it in this "if" statement. 
So, you want to say that we should make "lhs" and "rhs" constants here, right?